torch/quantization/observer.py

```diff
@@ -1072,6 +1073,91 @@ def forward(self, x):
+        suffix = mod._c.qualified_name.split('.', 1)[1]
+        name = re.sub(r'\.___torch_mangle_\d+', '', suffix)
+        return obs_type_name in name
+    return False
+
+def _is_activation_post_process(module):
+    return (isinstance(module, torch.quantization.ObserverBase) or
+            isinstance(module, torch.quantization.FakeQuantize) or
+            _is_observer_script_module(module, 'torch.quantization.observer'))
+
+def _is_per_channel_obs_instance(module):
+    if isinstance(module, torch.jit.RecursiveScriptModule):
+        return _is_observer_script_module(module, "torch.quantization.observer.PerChannelMinMaxObserver") or\
+            _is_observer_script_module(module, "torch.quantization.observer.MovingAveragePerChannelMinMaxObserver")
+    else:
+        return isinstance(module, torch.quantization.PerChannelMinMaxObserver) or\
+            isinstance(module, torch.quantization.MovingAveragePerChannelMinMaxObserver)
+
+def _is_per_tensor_obs_instance(module):
+    if isinstance(module, torch.jit.RecursiveScriptModule):
+        return _is_observer_script_module(module, "torch.quantization.observer.MinMaxObserver")
+    else:
+        return isinstance(module, torch.quantization.MinMaxObserver)
+
+def _is_hist_obs_instance(module):
+    if isinstance(module, torch.jit.RecursiveScriptModule):
+        return _is_observer_script_module(module, "torch.quantization.observer.MinMaxObserver")
+    else:
+        return isinstance(module, torch.quantization.HistogramObserver)
+
+def _get_attr_names_for_obs(module, name):
+    r"""
+    Return a dict that contains the module attr to fully qualified name
+    in the model. The name should correspond to the attr name in the model
+    state_dict.
+    """
+    qual_names = {}
+    qual_names['eps'] = name + '.eps'
+    if _is_per_channel_obs_instance(module):
+        qual_names['min_vals'] = name + '.min_vals'
+        qual_names['max_vals'] = name + '.max_vals'
+    elif _is_per_tensor_obs_instance(module):
+        qual_names['min_val'] = name + '.min_val'
+        qual_names['max_val'] = name + '.max_val'
+    elif _is_hist_obs_instance(module):
+        qual_names['histogram'] = name + '.histogram'
+        qual_names['min_val'] = name + '.min_val'
+        qual_names['max_val'] = name + '.max_val'
+
+    return qual_names
+
+def get_observer_state_dict(mod):
+    r"""
+    Returns the state dict corresponding to the observer stats.
+    Traverse the model state_dict and extract out the stats.
+    """
+    od = OrderedDict()
+    if isinstance(mod, torch.jit.RecursiveScriptModule):
+        for k, v in mod.state_dict().items():
+            if 'observer' in k:
+                od[k] = v
+    else:
+        # path for GraphModule and nn.Module (eager mode)
+        for k, v in mod.state_dict().items():
+            if 'activation_post_process' in k:
+                od[k] = v
+
+    return od
+
+def load_observer_state_dict(mod, obs_dict):
```

Comment:
instead of this, would it work to just load the state dict directly with strict==False, which will make it ignore missing keys for non-observer modules?  If that's possible, that would reduce the amount of custom code needed and eliminate another avenue for potential future bugs.

Reply:
We should also support a strict=False mode. Use case would be where we collect stats for all the modules, but quantize only a subset of them.

Reply:
strict=False should already be supported. I can make sure the test covers that case too. For the case you mentioned it would still not print warning/error messages when the subset that you want to quantize is missing from the state_dict. 
As an alternate for this scenario can we load all stats, but modify convert to accept a qconfig that will utilize only a subset of them?

Reply:
I'm just concerned that we are inventing one more paradigm about how to save/load things, which can grow to be a lot of baggage over time.  Can we avoid this entirely by using current state dicts as is, and creating fb-only functions to handle things like missing observers, changed module names, etc?

Reply:
I've updated the code to use a custom load_from_state_dict for PerChannelMinMaxObservers (since this observer requires resizing the min/max_vals to the right shape). For the rest it uses the default `load_from_state_dict`